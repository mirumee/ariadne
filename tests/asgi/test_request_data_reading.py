import json


def test_attempt_parse_request_missing_content_type_raises_bad_request_error(
    client, snapshot
):
    response = client.post("/", content="")
    assert response.status_code == 400
    snapshot.assert_match(response.text)


def test_attempt_parse_non_json_request_raises_bad_request_error(client, snapshot):
    response = client.post("/", content="", headers={"content-type": "text/plain"})
    assert response.status_code == 400
    snapshot.assert_match(response.text)


def test_attempt_parse_non_json_request_body_raises_bad_request_error(client, snapshot):
    response = client.post(
        "/", content="", headers={"content-type": "application/json"}
    )
    assert response.status_code == 400
    snapshot.assert_match(response.text)


def test_attempt_parse_json_scalar_request_raises_graphql_bad_request_error(
    client, snapshot
):
    response = client.post("/", json="json string")
    assert response.status_code == 400
    snapshot.assert_match(response.text)


def test_attempt_parse_json_array_request_raises_graphql_bad_request_error(
    client, snapshot
):
    response = client.post("/", json=[1, 2, 3])
    assert response.status_code == 400
    snapshot.assert_match(response.text)


def test_multipart_form_request_fails_if_operations_is_not_valid_json(client, snapshot):
    response = client.post(
        "/",
        data={
            "operations": "not a valid json",
            "map": json.dumps({"0": ["variables.file"]}),
        },
        files={"0": ("test.txt", "hello".encode("utf-8"))},
    )
    assert response.status_code == 400
    snapshot.assert_match(response.content)


def test_multipart_form_request_fails_if_map_is_not_valid_json(client, snapshot):
    response = client.post(
        "/",
        data={
            "operations": json.dumps(
                {
                    "query": "mutation($file: Upload!) { upload(file: $file) }",
                    "variables": {"file": None},
                }
            ),
            "map": "not a valid json",
        },
        files={"0": ("test.txt", "hello".encode("utf-8"))},
    )
    assert response.status_code == 400
    snapshot.assert_match(response.content)
