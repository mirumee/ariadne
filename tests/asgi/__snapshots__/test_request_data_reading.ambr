# serializer version: 1
# name: test_attempt_parse_json_array_request_raises_graphql_bad_request_error
  '{"errors":[{"message":"Operation data should be a JSON object"}]}'
# ---
# name: test_attempt_parse_json_scalar_request_raises_graphql_bad_request_error
  '{"errors":[{"message":"Operation data should be a JSON object"}]}'
# ---
# name: test_attempt_parse_non_json_request_body_raises_bad_request_error
  'Request body is not a valid JSON'
# ---
# name: test_attempt_parse_non_json_request_raises_bad_request_error
  'Posted content must be of type application/json or multipart/form-data'
# ---
# name: test_attempt_parse_request_missing_content_type_raises_bad_request_error
  'Posted content must be of type application/json or multipart/form-data'
# ---
# name: test_multipart_form_request_fails_if_map_is_not_valid_json
  b"Request 'map' multipart field is not a valid JSON"
# ---
# name: test_multipart_form_request_fails_if_operations_is_not_valid_json
  b"Request 'operations' multipart field is not a valid JSON"
# ---
